GNB 클론 디자인 적용

callme-waffle opened the pull request:
* 필수 모듈 설치
* recoil 사용을 위한 기본 directory 구성 및 적용 (/recoil/state.js & App.js)
* 이미치 추가 ( /assets/icons/* )
* GNB 레이아웃 디자인 일부 적용 ( /components/GlobalNavigationBar.jsx )
* SearchBar 사용목적 FilterArea 디자인 일부 적용 ( /components/FilterArea.jsx )
+) 상태처리관련 로직 적용되지 않음 -> 별도 PR 예정

callme-waffle committed:
Initialize Project

* 프로젝트 내 컴포넌트 구조 설계 (/construct/structure.md)
* 프로젝트 생성 (/project)

callme-waffle committed:
GNB 클론 디자인

* 필수 모듈 설치
* recoil 사용을 위한 기본 directory 구성 및 적용 (/recoil/state.js & App.js)
* 이미치 추가 ( /assets/icons/* )
* GNB 레이아웃 디자인 일부 적용 ( /components/GlobalNavigationBar.jsx )
* SearchBar 사용목적 FilterArea 디자인 일부 적용 ( /components/FilterArea.jsx )

Coreight98 commented on project/src/components/FilterArea.jsx:
```diff
@@ -0,0 +1,43 @@
+            gridTemplateColumns: ( type === "check-date" ) && `40% 60%`
+        }}
+    > 
+        {
```
삼항연산자를 사용하시기 보다는 가독성과 재사용성을 위해, props로 넘겨주시고, 조건문 대신 date,price,personal component로 만들어 넣는것은 어떨까요??
```
<div
      id={id || null}
      className={"filter-area " + (className || "")}
      onClick={onClick}
      style={{
        gridTemplateColumns: type === "check-date" && `40% 60%`,
      }}
    >
      {Children}
      {split !== false && <div className="filter-split-bar" />}
    </div>
```

```
<div className="search-bar-innerwrap">
            <FilterArea> 
                     <DateArea props={props}/>
            </FilterArea>
        </div>
```

Coreight98 commented on project/src/components/FilterArea/FilterBlock.jsx:
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect, useRef } from 'react';
+
+// css
+import "../../css/FilterBlock.css";
```
alias 같은 절대 경로도 도입해보시면 좋을거 같습니다!

Coreight98 reviewed (commented):
코드 작성하시느라 고생 많으셨습니다.
설계도 잘 나누신거 같고, 전체적으로 아토믹 구조처럼 잘게 자르려는 노력을 많이 하신거 같다고 생각합니다.😆😆👏 
좋은 코드 잘 봤습니다!

callme-waffle committed:
전역상태 & Calender 필터 추가

* 컴포넌트 이름 변경 ( PropsControl -> FilterControl ) (/*)
* stylesheet 경로 변경 ( /css/* -> /styles/* )
* Recoil 기반 Global States 선언 (/recoil/state.js)
* 상태 변경에 대한 css keyframe 애니메이션 추가 (/*)
* 웹팩기반 학습내용 추가 (/webpack/* | deprecated)
* FilterControl 활용목적 Calender 추가 (개발완료) (/components/GNB/FilterControl/Calender.jsx)
ㄴ 디자인: 월 | 요일 | 날짜 | 체크인/아웃 날짜 선택 | 월별 이동 화살표 | 1달 이상의 선택에 대한 배경 그라데이션
ㄴ 기능: 월 변경 | 체크인/아웃 날짜 선택 | 초기화
